backward.hpp

```diff
@@ -3354,8 +3381,28 @@ class SourceFile {
   typedef std::vector<std::pair<unsigned, std::string>> lines_t;
 
   SourceFile() {}
-  SourceFile(const std::string &path)
-      : _file(new std::ifstream(path.c_str())) {}
+  SourceFile(const std::string &path) {
+    // 1. If BACKWARD_CXX_SOURCE_PREFIXES is set then assume it contains
+    //    a colon-separated list of path prefixes.  Try prepending each
+    //    to the given path until a valid file is found.
+    const char* prefixes_str = std::getenv("BACKWARD_CXX_SOURCE_PREFIXES");
+    if (prefixes_str && !std::string(prefixes_str).empty()) {
+      std::vector<std::string> prefixes;
+      details::split_string(std::string(prefixes_str),
```

Comment:
The split operation will be performed every time a source file is opened. Even though the environment variable never changes. Not sure how easy it would be to cache this operation though. Let me know what you think.

Reply:
Done.  That's a good suggestion -- now it is cached in a function static variable and only computed once.